Slingshot.Api/Abstract/Repository.cs

```diff
@@ -145,15 +145,15 @@ public virtual Task<bool> HasScmInfo()
         {
             return Task.FromResult(false);
         }
-
-        public virtual Task WritePullRequestComment(string prId, string comment)
+        
+        public virtual Task<IPullRequestInfo> GetPullRequest(string prId)
         {
-            return Task.FromResult(new object());
+            return Task.FromResult<IPullRequestInfo>((IPullRequestInfo)new object());
         }
 
-        public virtual Task<T> GetPullRequest<T>(string prId) where T : IPullRequestInfo
+        public virtual Task UpdatePullRequest(string prId, dynamic content)
```

Comment:
Probably a bit cleaner to have it take an IPullRequestInfo. dynamic is an implementation detail of BitBucketPullRequest.